src/cli/java/org/commcare/util/screen/MultiSelectEntityScreen.java

```diff
@@ -134,9 +135,18 @@ private void prcessSelectionAsGuid(String guid) throws CommCareSessionException
                     "Could not make selection with reference id " + guid + " on this screen. " +
                             " If this error persists please report a bug to CommCareHQ.");
         }
+        validateEntitiesInInstance(cachedInstance);
         storageReferenceId = guid;
     }
 
+    private void validateEntitiesInInstance(ExternalDataInstance instance) throws CommCareSessionException {
```

Comment:
There is currently[ an issue](https://dimagi-dev.atlassian.net/browse/QA-5457) with this validation that I am verifying with delivery